File: frontend/src/services/conditions.ts
export const AVAILABLE_CONDITIONS = [
    { id: 'heat', label: '🔥 Heat (\u0394)' },
    { id: 'light', label: '☀️ Light (hv)' },
    { id: 'pd_c', label: '🔘 Pd/C' },
    { id: 'lindlar', label: '🍂 Lindlar Catalyst' },
    { id: 'cold', label: '❄️ Cold (-78°C)' },
    { id: 'high_concentration', label: '🧪 High Concentration' },
] as const;

export type ConditionId = typeof AVAILABLE_CONDITIONS[number]['id'];

export const QUICK_ADD_MOLECULES: Record<string, { smiles: string, label: string, isCondition: boolean }> = {
    'h2so4': { smiles: '[OH]S(=O)(=O)[OH]', label: '🧪 H₂SO₄', isCondition: false },
    'oh': { smiles: '[OH-]', label: '🧼 OH⁻', isCondition: false },
    'h2o': { smiles: 'O', label: '💧 H₂O', isCondition: false },
    'kmno4': { smiles: '[O-][Mn](=O)(=O)=O', label: '🧪 KMnO₄', isCondition: true },
    'mcpba': { smiles: 'O=C(OO)c1cccc(Cl)c1', label: '🧪 mCPBA', isCondition: true },
    'o3': { smiles: '[O-][O+]=O', label: '🔵 O₃', isCondition: true },
    'FeBr3': { smiles: '[Fe](Br)(Br)Br', label: '🧪 FeBr₃', isCondition: true },
    'FeCl3': { smiles: '[Fe](Cl)(Cl)Cl', label: '🧪 FeCl₃', isCondition: true },
    'AlCl3': { smiles: '[Al](Cl)(Cl)Cl', label: '🧪 AlCl₃', isCondition: true },
    'NaBH4': { smiles: '[Na+].[BH4-]', label: '🧪 NaBH₄', isCondition: true },
    'h3o': { smiles: '[OH3+]', label: '🧪 H₃O⁺', isCondition: true },
    'LiAlH4': { smiles: '[Li+].[AlH4-]', label: '🧪 LiAlH₄', isCondition: true },
    'NaH': { smiles: '[NaH]', label: '🧪 NaH', isCondition: true },
    'Ether': { smiles: 'O(CC)CC', label: '🧪 Ether', isCondition: true },
    'socl2': { smiles: 'ClS(=O)Cl', label: '🧪 SOCl₂', isCondition: true },
    'pbr3': { smiles: 'BrP(Br)Br', label: '🧪 PBr₃', isCondition: true },
    'pbcl': { smiles: 'ClP(Cl)Cl', label: '🧪 PCl₃', isCondition: true },
    'pcc': { smiles: '[nH+]1ccccc1.[O-][Cr](=O)(=O)Cl', label: '🧪 PCC', isCondition: true },
    'H2CrO4': { smiles: 'O[Cr](=O)(=O)O', label: '🧪 H₂CrO₄', isCondition: true },
    'H2O2': { smiles: 'OO', label: '🧪 H₂O₂', isCondition: true },
};

File: backend/reactions/conditions.py
# Condition ID/Label -> SMILES
# Used to augment reactants during matching
CONDITION_MOLECULES = {
    "h2so4": "OS(=O)(=O)O",
    "acid": "OS(=O)(=O)O",  # Generic acid often mapped to Sulfuric Acid for reaction matching purposes
    "br2": "BrBr",
    "cl2": "ClCl",
    "h2o": "O",
    "water": "O",
    "hbr": "Br",
    "hcl": "Cl",
    "hi": "I",
    "hf": "F",
    "socl2": "ClS(=O)Cl",
    "pbr3": "BrP(Br)Br",
    "pcl3": "ClP(Cl)Cl",
    "boh3": "B",  # Borane BH3
    "bh3": "B",
    "mcpba": "O=C(OO)c1cccc(Cl)c1",  # mCPBA approximation or generic peroxyacid
    # Use generic peroxyacid structure if needed: CC(=O)OO
    # But rule expects: [CX3](=[OX1])[OX2][OX2H1]
    "peroxyacid": "CC(=O)OO",
    "kmno4": "[O-][Mn](=O)(=O)=O",
    "h2": "[H][H]",
    "lindlar": "[H][H]",  # Lindlar is H2 + Catalyst, so provides H2
    "nanh2": "[Na+].[NH2-]",
    "nh2-": "[NH2-]",
    "base": "[OH-]",  # Generic base
    "oh-": "[OH-]",
    "mg": "[Mg]",
    "pcc": "Cl[Cr-](=O)(=O)O",  # PCC structure approximation or pyridinium chlorochromate
    # PCC Rule: [nH+]1ccccc1.[O-][Cr](Cl)(=O)=O
    # So we need both components or just the Cr part? The rule has 3 reactants: alcohol, pyridine, Cr part.
    # Let's map PCC to the Cr part + Pyridine
    # But dictionary key maps to single SMILES usually.
    # We can perform a split in matcher.
    "jones": "O[Cr](O)(=O)=O",  # Chromic acid H2CrO4
    "h2cro4": "O[Cr](O)(=O)=O",
    "nah": "[Na+].[H-]",
    "hydride": "[H-]",
    "lialh4": "[Li+].[AlH4-]",
    "nabh4": "[Na+].[BH4-]",
    "alcohol": "CO",  # Generic alcohol (Methanol) for checks?
    "ch3oh": "CO",
}
